keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1368 @@
+support ephemeral resources, an entry in the new PodSpec.Resources
+list can also be a ResourceClaimTemplate. When a Pod gets created,
+such a template will be used to create a normal ResourceClaim with the
+Pod as owner, and then the normal allocation of the resource takes
+place.
+
+For immediate allocation, scheduling Pods is simple because the
+resource is already allocated and determines the nodes on which the
+Pod may run. For delayed allocation, a node is selected tentatively
+and plugin(s) try to allocate their resources for that node. If that
+succeeds, the Pod can start to run. If it fails, the scheduler must
+determine whether some other node fits the requirements and if so,
+request allocation again. If no node fits because some resources were
+already allocated for a node and are only usable there, then those
+resources must be released and then get allocated elsewhere.
+
+The resources allocated for a ResourceClaim can be shared by multiple
+containers in a pod. Depending on the capabilities defined in the
+ResourceClaim by the plugin, a ResourceClaim can be used exclusively
+by one pod at a time, by a certain maximum number of pods, or an
+unlimited number of pods.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+Partial allocation.
+
+Pod requests GPU resource with 2Gb memory:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: device-consumer
+spec:
+  containers:
+  - name: my-ubuntu
+    image: ubuntu
+    command: ["/bin/program"]
+    resources:
+    - name: gpu_2Gb
+  resources:
+  - name: gpu_2Gb
+    template:
+      resourceClassName: "gpu"
+      parameters:
+        memory: "2Gb"
+```
+
+This request triggers partial resource allocation on the node that has
+a GPU device with 4Gb of memory for the duration of the execution of
+this Pod.
+
+If some other Pod uses 1Gb of that device, then 1Gb free GPU memory
+will left on that device after the resource is allocated. It can be
+used by another Pod requesting up to 1Gb of GPU memory.
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Scheduling is likely to be slower when many Pods request the new
+resource types, both because scheduling such a Pod involves more
+round-trips through the API server for ResourceClaimStatus updates and
+because scheduling one Pod may affect other Pods in ways that cannot
+be anticipated by the kube-scheduler. When many Pods compete for
+limited resources, multiple attempts may be needed before a suitable
+node is found.
```

Comment:
We can think about ways to mitigate the scheduling conflict.

Reply:
@cathyhongzhang in theory yes, but that would be visible only in cases of big clusters with many device requests.
With current set of hardware and overall accelerator usages, percentage of accelerated workloads is really small, thus should be not critical for next several years, until accelerators would become more common thing.

Reply:
@elezar: do you agree? Is the approach proposed in this KEP going to be useful for how NVIDIA hardware is used in clouds?